### ISSUE TITLE
Path for custom nodes changes when specifying a path for the generated diagram

### ISSUE THREAD

**Issue by da-edra** [mingrammer/diagrams#375]
Hello, first of all thank you so much for creating and maintaining this project, I love it! ❤️ 

I was working on a diagram where I store my diagram inside a folder named `img` and I'm also creating a Custom node to load Slack's logo from the `img` folder and my Python script is at the same level as the `img` folder, it looks like this:

```
.
├── diagram.py
├── img
│   ├── diagram.png
│   ├── slack_logo.png
```

However, when I try to load an image from my local storage I noticed that the path has changed from the path where my Python script is located to the `img`.

I expect that the path for custom images would be the same path as where my Python script is located, not the path where the image is generated, the reason for this is that it can be a little bit confusing if the path changes unexpectedly (like in my case).

Another use to keep the same path as where the script is located is for users who want to store their diagrams into a folder called `diagrams` and have some custom images in another folder named `logos` (both at the same level where their diagram is located).

I'm attaching a code snippet that demonstrates the problem as well as the generated diagram:

```python3
from diagrams import Diagram
from diagrams.custom import Custom
from PIL import Image

with Diagram("", show=False, filename="img/diagram"):
    Custom("Image loads", "slack_logo.png") - Custom("Image doesn't load", "img/slack_logo.png")
```

![diagram](https://user-images.githubusercontent.com/10160626/98628625-9f8e8880-22dc-11eb-95c8-5127b8b87b05.png)

**Comment by clayms** [comment]
First off, `Slack` is in `diagrams.saas.chat`.

```
from diagrams import Diagram
from diagrams.saas.chat import Slack

with Diagram("Slack", show=False):
    Slack("")
```

![image](https://user-images.githubusercontent.com/12793783/98684628-641fa880-2334-11eb-9a6f-e471764aacdc.png)



To the other part of your question, I could not replicate the same issue.  I would try adding a `./` to the beginning of the path. See below.
```
from diagrams import Diagram
from diagrams.custom import Custom

with Diagram("\nCustom Image", show=False):# as diag:
    same_folder = Custom("loads_1", "jenkins_docker.png")
    diff_folder = Custom("loads_2", "./img/jenkins_docker.png")
    same_folder - diff_folder
```
![image](https://user-images.githubusercontent.com/12793783/98686259-5703b900-2336-11eb-85b8-684f78541448.png)


You could also try the `Node` class directly.  It has the added benefit of giving access to all the other Graphviz Node attributes.
See: https://graphviz.gitlab.io/doc/info/attrs.html

Example code and output below:
```
from diagrams import Diagram, Node

with Diagram("\nCustom Image", show=False):
    label_spacer = '\n' * 5

    same_folder = Node(f"{label_spacer}loads_1", labelloc="c", image="jenkins_k8s.png", penwidth="0.0",
        height="0.8")
    diff_folder = Node("loads_2", image="./img/jenkins_k8s.png", penwidth="0.0")

    same_folder - diff_folder
```


![image](https://user-images.githubusercontent.com/12793783/98687919-46ecd900-2338-11eb-93fb-572066d09535.png)

**Comment by da-edra** [comment]
The Slack icon was not in the documentation.

Also, you're missing the `filename="img/diagram"` which I mentioned in the issue.

**Comment by clayms** [comment]
Yes, there are several issues raised about documentation and improving the ability to find the 'resource' you want to use.  I found cloning the repo then opening up the `/resources/` folder to be the best work-around to find out if what I want is already available.

Similar raised issues: 

https://github.com/mingrammer/diagrams/issues/78

https://github.com/mingrammer/diagrams/issues/354


_________

The following works with the `filename` parameter.  Note the use of `../`.

```
with Diagram("\nCustom Image", show=False, filename="./img/_test_file"):
    same_folder = Custom("same", "../jenkins_docker.png")
    diff_folder = Custom("diff", "../img/jenkins_docker.png")

    same_folder - diff_folder
```

![image](https://user-images.githubusercontent.com/12793783/98724551-ebcfdc00-2361-11eb-8b44-9eb97a72f759.png)



See:
https://github.com/mingrammer/diagrams/blob/dfd8e0a52c8c4d1c3ce95dc7161c23bb2eaf0acb/diagrams/__init__.py#L111
and 
https://graphviz.readthedocs.io/en/stable/api.html#graphviz.Graph.render
Specifically:
> The layout command is started from the directory of filepath, so that references to external files (e.g. [image=...]) can be given as paths relative to the DOT source file.

**Comment by da-edra** [comment]
I think I'm not explaining myself correctly, but I should not need to `cd` one directory up for the image to work.

From the documentation you linked it says `references to external files (e.g. [image=...]) can be given as paths relative to the DOT source file.` this means that if my dot source file is in `/here` and my images are in `/here/img` I should reference my images like `image=img/dog.png`, not like `dog.png`. I also shouldn't need to move one directory up for my DOT source file to read an image that is at the same level as my DOT source file.

That same thing should apply to `diagrams`, it should use the path where my Python file is located, like I said in my first comment, _I expect that the path for custom images would be the same path as where my Python script is located, not the path where the image is generated_.

**Comment by clayms** [comment]
This library calls the graphviz libraries `Digraph` function with the parameters `name=self.name` and `filename=self.filename`.
`self.dot` is the "`DOT source file`", and it is `self.filename` is the name of the DOT file.  When graphviz `render` function is called it is relative to the location of the DOT source file.  Which in your case is in a subdirectory.

If you don't specify the filename parameter, then graphviz "defaults to name + '.gv'".  

See:
https://graphviz.readthedocs.io/en/stable/api.html#graphviz.Digraph

This library deletes that file after it renders.
See: 
https://github.com/mingrammer/diagrams/blob/dfd8e0a52c8c4d1c3ce95dc7161c23bb2eaf0acb/diagrams/__init__.py#L148-L152

**Comment by vgropp** [comment]
As i ran into this issue with the path relative to the outfile and not to the source.py, maybe include a small hint into the documentation?
